Rework resource guide

Legend-Master opened the pull request:
<!-- Thank you for opening a PR! We really appreciate you taking the time to help out 🙌 -->

#### Description
<!-- Translators, try to follow this pattern when naming your PRs:
"i18n(language code): (short description)" -->
- What does this PR change? Give us a brief description. <!-- If it's an update try adding the commits as reference -->
- Closes #3479 <!-- Add an issue number if this PR will close it or remove it. -->

<!--
Here's what will happen next:

1. Our GitHub bots will run to check your changes.
   If they spot any broken links you will see some error messages on this PR.
   Don't hesitate to ask any questions if you're not sure what these mean!

2. In a few minutes, you'll be able to see a preview of your changes on Netlify 🥳

3. One or more of our maintainers will take a look and may ask you to make changes.
   We try to be responsive, but don't worry if this takes a day or two.

4. Reach out to us on Discord with any questions along the way:
   https://discord.com/invite/tauri
-->

I have changed the explanation about different syntaxes to be examples explained with inline documentations, the new documentations also contain example target paths, and I hope it makes it easier to understand what's going on

I kinda feel like this is too much clutter but I don't know if there's a better way, maybe separate a basic guide and an advanced guide? Would love some feedbacks

Also added a notice about the special case for Android

Legend-Master committed:
Rework resource guide

FabianLars commented on src/content/docs/develop/resources.mdx:
````diff
@@ -5,46 +5,69 @@ i18nReady: true
 }
 ```
 
-:::note
````
Unless this is not true anymore, please do not remove it as it was added after users had issues with it.

(The diff is super hard to read on mobile so i can't tell if it was integrated somewhere else)

Legend-Master replied:
Since we already have a note about `fs:allow-resource-read-recursive` below and we have explained where the files will be going, I think it's clear enough that when doing the scope thing, a recursive one should be used, and if people still can't get it, I think we should spend a bit on the scope documentations (e.g. the fs scope capability's description) instead

The old notice here feels a bit out of place, since we didn't mention anything permission related at that point, unlike our v1 doc page https://v1.tauri.app/v1/guides/building/resources/

lucasfernog replied:
maybe add a permissions section here instead? i feel like removing it makes it too easy for people to misconfigure the capabilities

Legend-Master replied:
I think a separate section makes sense, I could add that tomorrow

Legend-Master committed:
More docs for mapping

Legend-Master committed:
Change vscode default mdx formatter to prettier

Legend-Master committed:
Resolve resource file paths section

Legend-Master committed:
Expand Android example

Legend-Master committed:
Format settings.json

Legend-Master committed:
Small tweak

Legend-Master committed:
Prettier??

lucasfernog reviewed (approved):
pretty good overall

Legend-Master committed:
Add permissions section

Legend-Master committed:
Simplify opener one

Legend-Master committed:
Add simple tab and hide explanation in a tab

Legend-Master commented:
Changes:

- Added a permission section
- Added a simple tab to give people a sense on how it looks like and pushed the complex explanation to a separate tab

FabianLars commented on src/content/docs/develop/resources.mdx:
```diff
@@ -3,49 +3,130 @@ title: Embedding Additional Files
+</TabItem>
+</Tabs>
+
+To learn about where does `$RESOURCE` resolve to on each platforms, see the documentation of [`resource_dir`]
```
```suggestion
To learn about where `$RESOURCE` resolves to on each platforms, see the documentation of [`resource_dir`]
```

FabianLars commented on src/content/docs/develop/resources.mdx:
```diff
@@ -56,19 +137,111 @@ In the following explanations "target resource directory" is either the value af
+### Android
+
+Currently the resources are stored in the APK as assets so the return value of those APIs are not normal file system paths,
+we use a special URI prefix `asset://localhost/` here that can be used with the [`plugin-fs`],
+with that, you can read the files through [`FsExt::fs`] like this:
```
```suggestion
we use a special URI prefix `asset://localhost/` here that can be used with the [`fs` plugin].
With that plugin, you can read the files through [`FsExt::fs`] like this:
```

FabianLars commented on src/content/docs/develop/resources.mdx:
````diff
@@ -56,19 +137,111 @@ In the following explanations "target resource directory" is either the value af
+let json = app.fs().read_to_string(&resource_path);
+```
+
+If you want or must need the resource files to be on a real file system, copy the contents out manually through the [`plugin-fs`]
````
```suggestion
If you want or must have the resource files to be on a real file system, copy the contents out manually through the [`fs` plugin]
```

FabianLars commented on src/content/docs/develop/resources.mdx:
````diff
@@ -135,11 +314,78 @@ const langDe = JSON.parse(await readTextFile(resourcePath));
+}
+```
+
+To use it with the [`plugin-opener`]:
````
```suggestion
To use it with the [`opener` plugin]:
```

FabianLars commented on src/content/docs/develop/resources.mdx:
```diff
@@ -56,19 +137,111 @@ In the following explanations "target resource directory" is either the value af
+### Android
+
+Currently the resources are stored in the APK as assets so the return value of those APIs are not normal file system paths,
+we use a special URI prefix `asset://localhost/` here that can be used with the [`plugin-fs`],
```
```suggestion
we use a special URI prefix `asset://localhost/` here that can be used with the [`fs` plugin],
```

FabianLars commented on src/content/docs/develop/resources.mdx:
```diff
@@ -94,14 +270,17 @@ tauri::Builder::default()
-## Accessing files in JavaScript
+### JavaScript
 
 For the JavaScript side, you can either use a command like the one above and call it through `await invoke('hello')` or access the files using the [`plugin-fs`]
```
```suggestion
For the JavaScript side, you can either use a command like the one above and call it through `await invoke('hello')` or access the files using the [`fs` plugin].
```
not sure if this suggestion works or if i missed some that don't show up in the diff.

Legend-Master replied:
Seems like you've cached everything, GitHub's website doesn't let me to commit the suggestions though, did it locally

FabianLars commented on src/content/docs/develop/resources.mdx:
````diff
@@ -135,11 +314,78 @@ const langDe = JSON.parse(await readTextFile(resourcePath));
+}
+```
+
+To use it with the [`plugin-fs`]:
````
```suggestion
To use it with the [`fs` plugin]:
```

FabianLars commented on src/content/docs/develop/resources.mdx:
```diff
@@ -135,11 +314,78 @@ const langDe = JSON.parse(await readTextFile(resourcePath));
+[`resolveResource`]: https://tauri.app/reference/javascript/api/namespacepath/#resolveresource
 [`app`]: https://docs.rs/tauri/latest/tauri/struct.App.html
 [`apphandle`]: https://docs.rs/tauri/latest/tauri/struct.AppHandle.html
 [`plugin-fs`]: /plugin/file-system/
```
```suggestion
[`fs` plugin]: /plugin/file-system/
```

Legend-Master commented on src/content/docs/develop/resources.mdx:
```diff
@@ -135,11 +314,78 @@ const langDe = JSON.parse(await readTextFile(resourcePath));
 [basic setup]: /plugin/file-system/#setup
 [Scope Permissions]: /plugin/file-system/#scopes
 [scopes]: /plugin/file-system/#scopes
+[`plugin-opener`]: /plugin/opener/
```
```suggestion
[`opener` plugin]: /plugin/opener/
```

Legend-Master committed:
Apply suggestions from code review

Co-authored-by: FabianLars <FabianLars@users.noreply.github.com>

FabianLars committed:
i missed one in the review
